src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts

```diff
@@ -0,0 +1,367 @@
+        Utils.handleError(JSON.stringify(response),
+          this.messageService, 'settings.pricing.pricing_definition_creation_error');
+      }
+    }, (error) => {
+      this.spinnerService.hide();
+      Utils.handleHttpError(error, this.router, this.messageService, this.centralServerService, 'settings.pricing.pricing_definition_creation_error');
+    });
+  }
+
+  public updatePricingDefinition(pricingDefinition: PricingDefinition) {
+    this.spinnerService.show();
+    this.centralServerService.updatePricingDefinition(pricingDefinition).subscribe((response: ActionResponse) => {
+      this.spinnerService.hide();
+      if (response.status === RestResponse.SUCCESS) {
+        this.messageService.showSuccessMessage('settings.pricing.pricing_definition_update_success');
+        this.closeDialog(true);
+      } else {
+        Utils.handleError(JSON.stringify(response),
+          this.messageService, 'settings.pricing.pricing_definition_update_error');
+      }
+    }, (error) => {
+      this.spinnerService.hide();
+      switch (error.status) {
+        case HTTPError.OBJECT_DOES_NOT_EXIST_ERROR:
+          this.messageService.showErrorMessage('settings.pricing.pricing_definition_not_found');
+          break;
+        default:
+          Utils.handleHttpError(error, this.router, this.messageService,
+            this.centralServerService, 'settings.pricing.pricing_definition_update_error');
+      }
+    });
+  }
+
+  public refresh() {
+    this.loadPricing();
+  }
+
+  public setMinDate(event) {
+    this.minDate = moment(event.value).toDate();
+  }
+
+  private initializeDimensions() {
+    for (const dimensionKey of this.dimensionsKeys) {
+      this[`${dimensionKey}Enabled`].setValue(this.dimensionsMap[dimensionKey].active);
+      this[`${dimensionKey}Value`].setValue(this.dimensionsMap[dimensionKey].price);
+      if (!!this.dimensionsMap[dimensionKey].stepSize) {
+        this[`${dimensionKey}StepEnabled`].setValue(true);
+        this[`${dimensionKey}StepValue`].setValue(this.dimensionsMap[dimensionKey].stepSize);
+      }
+    }
+  }
+
+  private consistencyCheck(pricingDefinition: PricingDefinition) {
+    if (!this.connectorPowerEnabled.value) {
+      delete pricingDefinition.staticRestrictions.connectorPowerkW;
+    }
+    if (!this.parkingTimeEnabled.value || !this.parkingTimeStepEnabled.value) {
+      delete pricingDefinition.dimensions.parkingTime.stepSize;
+    }
+    if (!this.chargingTimeEnabled.value || !this.chargingTimeStepEnabled.value) {
+      delete pricingDefinition.dimensions.chargingTime.stepSize;
+    }
+    if (!this.energyEnabled.value || !this.energyStepEnabled.value) {
+      delete pricingDefinition.dimensions.energy.stepSize;
+    }
+    for (const dimensionKey in pricingDefinition.dimensions) {
+      if (!pricingDefinition.dimensions[dimensionKey].active) {
+        delete pricingDefinition.dimensions[dimensionKey].price;
+      }
+    }
+    if (this.connectorType.value === 'A') {
```

Comment:
Use a constant here, it can be one constant that we can use everywhere not only for the connectors.

Reply:
will commit change ✅